qiskit/transpiler/__init__.py

```diff
@@ -40,629 +41,787 @@
+in computing the best mapping.  Most circuits will have a distribution of depths,
+perhaps not as wide as this one, due to the stochastic nature of the default SWAP
+mapper. Of course, we want the best circuit we can get, especially in cases where
+the depth is critical to success or failure. In cases like this, it is best to
+:func:`transpile` a circuit several times, e.g. 10, and take the one with the
+lowest depth. The :class:`~.SabreSwap` pass will do this by default by running its
+algorithm in parallel with multiple seed values and selecting the output which
+used the least number of SWAPs. If you would like to increase the number of trials
+:class:`~.SabreSwap` runs you can refer to :ref:`working_with_preset_pass_managers`
+and modifying the ``routing`` stage with a custom instance of :class:`~.SabreSwap`
+with a larger value for the ``trials`` argument.
 
-      The output from :func:`transpile` varies due to the stochastic swap mapper.
-      So the numbers below will likely change each time you run the code.
+.. _optimization_stage:
 
+Optimization Stage
+------------------
 
-   .. jupyter-execute::
+Decomposing quantum circuits into the basis gate set of the target device,
+and the addition of SWAP gates needed to match hardware topology, conspire to
+increase the depth and gate count of quantum circuits.  Fortunately many routines
+for optimizing circuits by combining or eliminating gates exist.  In some cases
+these methods are so effective the output circuits have lower depth than the inputs.
+In other cases, not much can be done, and the computation may be difficult to
+perform on noisy devices.  Different gate optimizations are turned on with
+different ``optimization_level`` values.  Below we show the benefits gained from
+setting the optimization level higher:
 
-      import matplotlib.pyplot as plt
-      from qiskit import QuantumCircuit, transpile
-      from qiskit.providers.fake_provider import FakeBoeblingen
-      backend = FakeBoeblingen()
+.. important::
 
-      ghz = QuantumCircuit(5)
-      ghz.h(0)
-      ghz.cx(0,range(1,5))
-      ghz.draw(output='mpl')
+   The output from :func:`transpile` varies due to the stochastic swap mapper.
+   So the numbers below will likely change each time you run the code.
 
 
-   .. jupyter-execute::
+.. jupyter-execute::
 
-      for kk in range(4):
-         circ = transpile(ghz, backend, optimization_level=kk)
-         print('Optimization Level {}'.format(kk))
-         print('Depth:', circ.depth())
-         print('Gate counts:', circ.count_ops())
-         print()
+   import matplotlib.pyplot as plt
+   from qiskit import QuantumCircuit, transpile
+   from qiskit.providers.fake_provider import FakeBoeblingen
+   backend = FakeBoeblingen()
 
+   ghz = QuantumCircuit(5)
+   ghz.h(0)
+   ghz.cx(0,range(1,5))
+   ghz.draw(output='mpl')
 
-   .. raw:: html
 
-      <br>
+.. jupyter-execute::
+
+   for kk in range(4):
+      circ = transpile(ghz, backend, optimization_level=kk)
```

Comment:
Perhaps we can change the `kk` stuff into `level`.

Reply:
Updated in: https://github.com/Qiskit/qiskit-terra/pull/9087/commits/a46e97de3261a7a394c0f63ff23f818f5936f49f